libraries/botbuilder-ai/botbuilder/ai/luis/luis_recognizer_v3.py

```diff
@@ -0,0 +1,176 @@
+from .luis_recognizer_options_v3 import LuisRecognizerOptionsV3
+from .luis_application import LuisApplication
+
+from botbuilder.core import (
+    RecognizerResult,
+    TurnContext,
+)
+from .activity_util import ActivityUtil
+
+
+class LuisRecognizerV3(LuisRecognizerInternal):
+    _dateSubtypes = ["date", "daterange", "datetime", "datetimerange", "duration", "set", "time", "timerange"]
+    _geographySubtypes = ["poi", "city", "countryRegion", "continent", "state"]
+    _metadata_key = "$instance"
+    # The value type for a LUIS trace activity.
+    luis_trace_type: str = "https://www.luis.ai/schemas/trace"
+
+    # The context label for a LUIS trace activity.
+    luis_trace_label: str = "Luis Trace"
+
+    def __init__(self, luis_application: LuisApplication, luis_recognizer_options_v3: LuisRecognizerOptionsV3 = None):
+        super().__init__(luis_application)
+
+        self.luis_recognizer_options_v3 = luis_recognizer_options_v3 or LuisRecognizerOptionsV3()
+        self._application = luis_application
+
+    async def recognizer_internal(
+            self,
+            turn_context: TurnContext):
+        recognizer_result: RecognizerResult = None
+
+        utterance: str = turn_context.activity.text if turn_context.activity is not None else None
+
+        url = self._build_url()
+        body = self._build_request(utterance)
+        headers = {
+            'Ocp-Apim-Subscription-Key': self.luis_application.endpoint_key,
+            'Content-Type': 'application/json'
+        }
+
+        async with aiohttp.ClientSession() as session:
+            async with session.post(url, json=body, headers=headers) as result:
+                luis_result = await result.json()
+                recognizer_result["intents"] = self._get_intents(luis_result["prediction"])
+                recognizer_result["entities"] = self._extract_entities_and_metadata(luis_result["prediction"])
+
+        return recognizer_result
+
+    def _build_url(self):
+
+        base_uri = self._application.endpoint or 'https://westus.api.cognitive.microsoft.com';
+        uri = "%s/luis/prediction/v3.0/apps/%s" % (base_uri, self._application.application_id)
+
+        if (self.luis_recognizer_options_v3.version):
+            uri += "versions/%/predict" % (self.luis_recognizer_options_v3.version)
+        else:
+            uri += "slots/%/predict" % (self.luis_recognizer_options_v3.slot)
+
+        params = "?verbose=%s&show-all-intents=%s&log=%s" % (
+            "true" if self.luis_recognizer_options_v3.include_instance_data else "false",
+            "true" if self.luis_recognizer_options_v3.include_all_intents else "false",
+            "true" if self.luis_recognizer_options_v3.log else "false")
+
+        return uri + params
+
+    def _build_request(self, utterance: str):
+        body = {
+            'query': utterance,
+            'preferExternalEntities' : self.luis_recognizer_options_v3.prefer_external_entities
```

Comment:
Compare to JS which sets 'overridePredictions' with the value of self.luis_recognizer_options_v3.prefer_external_entities.  Not sure which is correct.

Reply:
This one is correct. Im addressing this in the same JS PR